### PR TITLE
Updated for rocm 6.1.1 and fixed a py test

### PR DESCRIPTION
**Pull request by ItsPitt**
All faiss and py tests are passing currently (1 py test skipped) on navi 31.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -4,17 +4,22 @@ ENV TZ=America/Chicago
...
 
 RUN apt upgrade -y
-RUN apt update && apt install -y sudo wget gnupg2 git gcc gfortran libboost-dev bzip2 openmpi-bin flex build-essential bison libboost-all-dev vim libsqlite3-dev numactl sqlite3 gdb
-RUN wget -q -O - https://repo.radeon.com/rocm/rocm.gpg.key | sudo apt-key add -
-RUN echo 'deb [arch=amd64] https://repo.radeon.com/rocm/apt/debian/ ubuntu main' | sudo tee /etc/apt/sources.list.d/rocm.list
-RUN apt update
-RUN apt install -y rocm-dev6.0.0 rocm-libs6.0.0 build-essential libssl-dev swig numpy-stl
+RUN apt update && apt install -y sudo wget gnupg2 git gcc gfortran libboost-dev bzip2 openmpi-bin flex build-essential bison libboost-all-dev vim libsqlite3-dev numactl sqlite3 gdb libgtest-dev
+
+#Rocm 6.1.1
```
Comment (iotamudelta) [inline Dockerfile]: ROCm, not Rocm :-)

File: Dockerfile [hunk 2]
```diff
@@ -4,17 +4,22 @@ ENV TZ=America/Chicago
...
+
+RUN apt update && apt install -y build-essential libssl-dev swig python3
+RUN pip install pytest scipy numpy==1.23.5
 
 COPY target.lst /opt/rocm/bin/
 
 # recent cmake required for FAISS 
 WORKDIR /root
-RUN wget https://github.com/Kitware/CMake/releases/download/v3.23.1/cmake-3.23.1.tar.gz && tar -zxvf cmake-3.23.1.tar.gz && cd cmake-3.23.1 && ./bootstrap && make -j && make install
+RUN wget https://github.com/Kitware/CMake/releases/download/v3.29.4/cmake-3.29.4.tar.gz && tar -zxvf cmake-3.29.4.tar.gz && cd cmake-3.29.4 && ./bootstrap && make -j && make install
```
Comment (iotamudelta) [inline Dockerfile]: why update? I deliberately kept it at the minimal supported version.

File: Dockerfile [hunk 3]
```diff
@@ -34,22 +39,26 @@ RUN ./configure --prefix=/opt/libflame --enable-lapack2flash --enable-vector-int
 RUN make -j
 RUN make install
 
-ENV LD_LIBRARY_PATH=/opt/rocm/lib:$LD_LIBRARY_PATH
+ENV LD_LIBRARY_PATH=/opt/rocm/lib
```
Comment (iotamudelta) [inline Dockerfile:42]: why?

**Review by iotamudelta — CHANGES_REQUESTED** [review]
minor stuff

File: Dockerfile (hunk 1 above)
Comment (ItsPitt) [reply Dockerfile]: Will fix

File: Dockerfile (hunk 2 above)
Comment (ItsPitt) [reply Dockerfile]: Updated when testing other stuff. Not needed. Will revert

File: Dockerfile (hunk 3 above)
Comment (ItsPitt) [reply Dockerfile:42]: The Docker build was giving a warning that there was no var already set

File: Dockerfile (hunk 1 above)
Comment (ItsPitt) [reply Dockerfile]: Fixed: e18bd4c176800c0ce3c246d380bdb788e2fb94a0

File: Dockerfile (hunk 2 above)
Comment (ItsPitt) [reply Dockerfile]: 3.23.1 no longer works. Must be 3.24.0+. Fixed: e18bd4c176800c0ce3c246d380bdb788e2fb94a0

**Review by iotamudelta — APPROVED** [review]
LGTM

**Pull request merged**